### DISCUSSION TITLE
blog/my-ultimate-terminal-customizations-for-macos

### DISCUSSION THREAD

**Discussion by giscus** [cesarsotovalero/cesarsotovalero.github.io#55] [Announcements]
# blog/my-ultimate-terminal-customizations-for-macos

There is a common belief that top-class software developers can do their job without ever abandoning the terminal. This blog post is a how-to guide for you (and my future self) to configure the terminal in a way so that we will enjoy every command typed on it.

https://www.cesarsotovalero.net/blog/my-ultimate-terminal-customizations-for-macos.html

**Comment by martisak** [comment]
Some really good tips here! Half of them were new to me, trying them out now!

Brew probably deserves its own spot here :). It also has a zsh-plugin with the same name.

I've been using iTerm2 for a long time, but recently switched to Kitty https://sw.kovidgoyal.net/kitty/. It feels faster, and has a bit more nerdy way of splitting the window. Since I am more of a Python "programmer", I use more Python-centric zsh-plugins such as pipenv.

But there is one thing that completely changed my terminal-life and that is fzf! https://github.com/junegunn/fzf
For one, you can fuzzy find stuff in the history, but it can be used for so many things. I use it to interact with our Kubernetes cluster, for example. Here is one command to find to describe a Kubernetes job:

```
function kdj() {
  kubectl describe job $(kubectl get jobs | sed 1d | fzf -1 -q "$1" | awk '{print $1}')
}
```

Another brilliant thing with this blog post are the examples in asciinema!

**Comment by martisak** [comment]
Here is another interesting tool that corrects errors in previous console commands. https://github.com/nvbn/thefuck

**Comment by rjalexa** [comment]
Very interesting. Unix sysadmin and python developer here. Spend a lot of time in the shell (zsh) on my Mac. Been using iTerm2 for a long time now switched to Hyper (and I don't rem why LOL). I am currently struggling transitioning from a brew/conda environment to a pyenv/poetry environment and still banging my head when things don't work. Wonder how to go through an intelligent PATH cleanup but be still be left with funcitoning poetry venvs, brew and conda without breaking them. Thanks
